### ISSUE TITLE
checkLoginIframe not working on mobile and tablet

### ISSUE THREAD

**Issue by Salvatore-Als** [keycloak/keycloak#12996]
### Describe the bug

Hello,
The problem is Keycloak initialization on phone/tablet (chrome, mozilla, safari), Keycloak does not initialize, because of the `checkLoginIframe: true`

```export function initializer(keycloak: KeycloakService): () => Promise<any> {
    return (): Promise<any> => {
        return new Promise(async (resolve, reject) => {
            console.log("Init from Keycloak Provider");

            keycloak.init({
                config: kcConfig as KeycloakConfig,
                initOptions: {
                    onLoad: 'login-required',
                    checkLoginIframe: true,
                    checkLoginIframeInterval: 60,
                    enableLogging: true,
                },
                enableBearerInterceptor: true,
                bearerPrefix: 'Bearer',
                bearerExcludedUrls: ['/assets'],
            }).then((value: any) => {
                console.log("Keycloak init", value);
                resolve(value);
            }).catch((error: any) => {
                console.log("Keycloak init error", error);
                reject(error);
            });
        });
    }
}

The problem does not come from the angular wrapper, my tests without the wrapper have the same problem.
We get the following error: **Timeout when waiting for 3rd party check iframe message**

Best Regards,


### Version

Keycloak JS 18.0.1

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

**Comment by CSalih** [comment]
Hi @Salvatore-Als, 
you may need to add your domain in the CSP headers (i guess `frame-src` and `frame-ancestors` is missing). You can find the CSP configuration here:  `Realm Settings` > `Security Defenses` > ` Content-Security-Policy`.
Regards

**Comment by Salvatore-Als** [comment]
Hello,
Thanks for your reply :) 

The frame-src and frame-ancestors is set, and this one works on Desktop :/ 

I have made severals test, it seems the iframe is not working on mobile/tablet device.

Best Regards,

**Comment by CSalih** [comment]
Hi, 
yes I can reproduce that same issue on the Safari Browser (Desktop and Mobile). I will get back to you when i found a solution.

**Comment by Salvatore-Als** [comment]
Hello,
Thanks for the feedback, I will check ASAP !

**Comment by mkumatag** [comment]
I experience this issue on safari browser while running react app connecting to keycloak 21.1.2, do we any workaround for this issue?

**Comment by jonkoops** [comment]
The issue in Safari is a known issue (#21307), it should be fixed with 22.0.0.

**Comment by jonkoops** [comment]
I am unable to reproduce this issue any longer, so I will assume it to be fixed. If anyone has a means of reproducing this issue, please let me know.
